From 817ff2859c62479528e82f040add61ac2ceab1d0 Mon Sep 17 00:00:00 2001
From: kashike <kashike@vq.lc>
Date: Thu, 15 May 2014 22:42:51 -0400
Subject: [PATCH] Expose CraftItemStack parent

---
 .../org/bukkit/craftbukkit/inventory/CraftItemStack.java     | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
index 23f05f4d..91d287b8 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
@@ -91,6 +91,11 @@ public final class CraftItemStack extends ItemStack {
     }
 
     net.minecraft.server.ItemStack handle;
+    // Magnet start
+    public net.minecraft.server.ItemStack parent() {
+        return this.handle;
+    }
+    // Magnet end
 
     /**
      * Mirror
-- 
2.18.0.windows.1

